### ISSUE TITLE
bcftools extracting consensus sequence

### ISSUE THREAD

**Issue by computingmachine** [samtools/bcftools#906]
Hi I am looking for the code to extract the consensus sequence and the website has the code below, which returns me different lengths from different individuals, I guess it is just because the bcftools consider SNPs and INDEL in the VCF file? Is there a way to consider only the SNPs in the consensus? In addition, only ATGC are shown in the sequence, is it possible to include the ambiguity code?

bcftools mpileup -Ou -f reference.fa alignments.bam | bcftools call -mv -Oz -o calls.vcf.gz
tabix calls.vcf.gz
cat reference.fa | bcftools consensus calls.vcf.gz > consensus.fa

**Comment by pd3** [comment]
You can add `-i 'type="snp"'` to include only SNPs and `-I` to output the abmiguity codes. Check also the usage page (type "`bcftools consensus`" without arguments) and the manual page http://samtools.github.io/bcftools/bcftools.html#consensus for more.

**Comment by balasink** [comment]
Hi I tried the above script but for some reason my final consensus sequence only has the sequence for my reference species.
I am trying to extract consensus sequences for about 30 species and I've been using the script each time for each species, maybe that's the problem? After bwa mem alignment and generating a sorted alignment bam file for each species I tried:

bcftools mpileup -Ou -f ref_geneA.fa species1.aln.sorted.bam | bcftools call -mv -Oz -o species1.calls.vcf.gz

tabix species1.calls.vcf.gz

cat ref_geneA.fa | bcftools consensus species1.calls.vcf.gz > consensus.fa

But the consensus.fa only has the ref_geneA sequence in it and nothing else from species1.
